File: java/121.best-time-to-buy-and-sell-stock.java
/*
 * @lc app=leetcode id=121 lang=java
 *
 * [121] Best Time to Buy and Sell Stock
 */
class Solution {
    public int maxProfit(int[] prices) {
        int buy = Integer.MAX_VALUE, profit = 0;
        for (int i = 0; i < prices.length; i++) {
            buy = Math.min(buy, prices[i]);
            profit = Math.max(profit, prices[i] - buy);
        }
        return profit;
    }
}

File: cpp/121_Best Time to Buy and Sell Stock.cpp
class Solution {
public:
	int maxProfit(vector<int>& prices) {
		int res = 0;
		int minPri = INT_MAX;
		for (int i = 0; i < prices.size(); i++) {
			minPri = min(minPri, prices[i]);
			res = max((prices[i] - minPri), res);
		}
		return res;
	}
};